authorization-common/src/main/java/com/sequenceiq/authorization/service/CommonPermissionCheckingUtils.java

```diff
@@ -18,39 +22,95 @@
+
+    private final UmsResourceAuthorizationService umsResourceAuthorizationService;
+
+    private final Map<AuthorizationResourceType, DefaultResourceChecker> defaultResourceCheckers;
 
     @Inject
-    private UmsResourceAuthorizationService umsResourceAuthorizationService;
+    public CommonPermissionCheckingUtils(UmsAccountAuthorizationService umsAccountAuthorizationService,
+            UmsResourceAuthorizationService umsResourceAuthorizationService,
+            List<DefaultResourceChecker> defaultResourceCheckers) {
+        this.umsAccountAuthorizationService = umsAccountAuthorizationService;
+        this.umsResourceAuthorizationService = umsResourceAuthorizationService;
+        this.defaultResourceCheckers = Maps.immutableEnumMap(defaultResourceCheckers.stream()
+                .collect(toMap(DefaultResourceChecker::getResourceType, Function.identity())));
+    }
 
     public void checkPermissionForUser(AuthorizationResourceType resource, AuthorizationResourceAction action, String userCrn) {
         umsAccountAuthorizationService.checkRightOfUser(userCrn, resource, action);
     }
 
     public void checkPermissionForUserOnResource(AuthorizationResourceType resource, AuthorizationResourceAction action, String userCrn, String resourceCrn) {
-        umsResourceAuthorizationService.checkRightOfUserOnResource(userCrn, resource, action, resourceCrn);
+        DefaultResourceChecker defaultResourceChecker = defaultResourceCheckers.get(resource);
+        if (defaultResourceChecker == null || !defaultResourceChecker.isDefault(resourceCrn)) {
+            umsResourceAuthorizationService.checkRightOfUserOnResource(userCrn, resource, action, resourceCrn);
+        } else {
+            if (!defaultResourceChecker.isAllowedAction(action)) {
+                String right = RightUtils.getRight(resource, action);
+                String msg = String.format("You have no right to perform %s on resource %s", right, resourceCrn);
+                LOGGER.error(msg);
+                throw new AccessDeniedException(msg);
+            }
+        }
     }
 
     public void checkPermissionForUserOnResources(AuthorizationResourceType resource, AuthorizationResourceAction action,
             String userCrn, Collection<String> resourceCrns) {
-        umsResourceAuthorizationService.checkRightOfUserOnResources(userCrn, resource, action, resourceCrns);
+        DefaultResourceChecker defaultResourceChecker = defaultResourceCheckers.get(resource);
+        if (defaultResourceChecker == null) {
+            umsResourceAuthorizationService.checkRightOfUserOnResources(userCrn, resource, action, resourceCrns);
+        } else {
+            Collection<String> defaultResourceCrns = defaultResourceChecker.getDefaultResourceCrns(resourceCrns);
+            Collection<String> notDefaultResourceCrns = resourceCrns.stream()
+                    .filter(crn -> !defaultResourceCrns.contains(crn))
+                    .collect(Collectors.toList());
```

Comment:
This also appears twice, maybe it is wort to move out into a method to increase readability of the code